src/Modal/Modal.component.tsx

```diff
@@ -259,25 +175,25 @@ export class Modal extends React.Component<Props> {
         isOpen={this.props.isOpen}
         style={castStyles}
         modalSize={this.props.modalSize || 'md'}
-        {...this.props.controlSpecificProps}
+        {...this.props}
       >
         {this.props.modalTitle &&
         <ModalHeaderDiv>
           <h5>{this.props.modalTitle}</h5>
-          <button type="button" aria-label="Close" onClick={() => this.closeModal(this.props.onCancelOrNo)}>
+          {this.props.onTitleClose &&
+          <button type="button" aria-label="Close"
+            onClick={() => null}>
```

Comment:
The `onTitleClose` prop is not being used as expected. 

I suppose we are to do something like `onClick={this.props.onTitleClose}`?